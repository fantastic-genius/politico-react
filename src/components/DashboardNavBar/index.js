import React from 'react';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import { signout } from '@actions/Auth';

const DashboardNavBar = (props) => {
  const onSignOut = () => {
    props.signout();
    <Redirect to='/' />
  }
  return (
    <header>
      <div className="title">
        <div className="app-name">
          <Link to='/'>
            <h2>POLITICO</h2>
          </Link>
        </div>
      </div>
      <div className="logout-con">
        <button id="logout" className="btn btn-warning" onClick={onSignOut}><a href="#">Logout</a></button>
      </div>
    </header>
  );
}
const mapStateToProps = state => ({
  user: state.auth.user
})
export default connect(mapStateToProps, {signout})(DashboardNavBar);
