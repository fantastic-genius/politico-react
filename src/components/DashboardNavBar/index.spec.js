import React from 'react';
import { shallow } from 'enzyme';
import DashboardNavBar from './index';

describe('<DashboardNavBar />', () => {
  describe('Rendering', () => {
    it('should render DashboardNavBar correctly in debug mode', () => {
        const component = shallow(<DashboardNavBar />);
        expect(component).toMatchSnapshot();
    });
  })

})

