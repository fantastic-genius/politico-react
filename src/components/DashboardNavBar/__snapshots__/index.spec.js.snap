// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardNavBar /> Rendering should render DashboardNavBar correctly in debug mode 1`] = `
<header>
  <div
    className="title"
  >
    <h3>
      User Dashboard
    </h3>
  </div>
  <div
    className="logout-con"
  >
    <button
      className="btn btn-warning"
      id="logout"
    >
      <a
        href="#"
      >
        Logout
      </a>
    </button>
  </div>
</header>
`;
